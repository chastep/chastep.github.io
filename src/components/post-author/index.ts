export { PostAuthor } from "./post-author";
