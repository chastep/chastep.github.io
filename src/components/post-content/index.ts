export { PostContent } from "./post-content";
