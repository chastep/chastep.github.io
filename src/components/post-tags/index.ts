export { PostTags } from "./post-tags";