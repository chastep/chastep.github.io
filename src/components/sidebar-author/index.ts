export { SidebarAuthor } from "./sidebar-author";