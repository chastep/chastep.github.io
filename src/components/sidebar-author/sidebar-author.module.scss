@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

$photo-size: 75px;

.sidebar-author {
  .photo {
    border: 1px solid hsl(var(--color-muted));
    background-clip: padding-box;
    display: inline-block;
    border-radius: 50%;
    height: $photo-size;
    width: $photo-size;
    @include mixins.margin-bottom(0);

    img {
      border-radius: 50%;
    }
  }

  .title {
    font-size: variables.$typography-base-font-size * 1.125;
    font-weight: 600;
    margin: 0;
    @include mixins.line-height(1.125);

    .link {
      color: hsl(var(--color-foreground));

      &:hover,
      &:focus {
        color: hsl(var(--color-foreground));
      }
    }
  }

  .description {
    color: hsl(var(--color-muted-foreground));
    @include mixins.line-height(1);
    @include mixins.margin-bottom(1);
  }

  .titleContainer {
    align-items: center;
    display: flex;
    justify-content: space-between;
    @include mixins.margin(0.5, 0, 0.5, 0);
  }
}
