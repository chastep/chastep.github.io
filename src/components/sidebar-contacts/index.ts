export { SidebarContacts } from "./sidebar-contacts";
