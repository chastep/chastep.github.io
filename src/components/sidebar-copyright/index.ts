export { SidebarCopyright } from "./sidebar-copyright";
