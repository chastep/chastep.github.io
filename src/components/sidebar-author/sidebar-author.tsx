import React, { type FC } from "react";
import { Link } from "gatsby";

import { Image } from "@/components/image";
import { ThemeSwitcher } from "@/components/theme-switcher";

import * as styles from "./sidebar-author.module.scss";

type SidebarAuthorProps = {
  author: {
    title: string;
    photo: string;
    description: string;
  };
  isHome?: boolean;
};

const SidebarAuthor: FC<SidebarAuthorProps> = ({ author, isHome }) => (
  <div className={styles.sidebarAuthor}>
    <Link to="/">
      <Image alt={author.title} path={author.photo} className={styles.photo} />
    </Link>

    <div className={styles.titleContainer}>
      {isHome ? (
        <h1 className={styles.title}>
          <Link className={styles.link} to="/">
            {author.title}
          </Link>
        </h1>
      ) : (
        <h2 className={styles.title}>
          <Link className={styles.link} to="/">
            {author.title}
          </Link>
        </h2>
      )}
      <ThemeSwitcher />
    </div>
    <p className={styles.description}>{author.description}</p>
  </div>
);

export { SidebarAuthor };
