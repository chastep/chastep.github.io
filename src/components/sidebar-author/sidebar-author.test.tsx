import React from "react";
import { StaticQuery } from "gatsby";
import { test, describe, expect, beforeEach, mock } from "bun:test";

import * as mocks from "@/mocks";
import { SidebarAuthor } from "@/components/sidebar-author";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

const mockedStaticQuery = mock((props) => <StaticQuery {...props} />);

describe("SidebarAuthor", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(() => <></>);
  });

  test("renders correctly", () => {
    const props = { isHome: false, author: mocks.author };
    const tree = createSnapshotsRenderer(<SidebarAuthor {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
