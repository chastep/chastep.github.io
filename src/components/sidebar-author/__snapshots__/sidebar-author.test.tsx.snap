// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarAuthor renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": null,
      "props": {
        "aria-current": "page",
        "className": "",
        "href": "/",
        "onClick": [Function],
        "onMouseEnter": [Function],
        "style": {},
      },
      "type": "a",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "John Doe",
              ],
              "props": {
                "aria-current": "page",
                "className": "",
                "href": "/",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "style": {},
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "h2",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
      ],
      "props": {
        "className": undefined,
      },
      "type": "p",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
